File: scala/src/main/scala/clase3/Split.scala
package clase3

object Split extends App {

  def split(list: List[Int]): (List[Int], List[Int]) = {

    def doSplit(list: List[Int], accL: List[Int], accR: List[Int]): (List[Int], List[Int]) = list match {
      case x :: y :: rest => doSplit(rest, accL :+ x, accR :+ y)
      case x :: Nil => (accL :+ x, accR)
      case _ => (accL, accR)
    }

    doSplit(list, Nil, Nil)

  }

  val l1 = List(10, 20, 30, 40, 50, 60, 70, 80, 90)
  val (l2, l3) = split(l1)

  println(l2)
  println(l3)

}

File: haskell/src/clase3/Split.hs
module Split where

doSplit :: [] Int -> [] Int -> [] Int -> ([] Int, [] Int)

doSplit (x:y:rest) accL accR = doSplit rest (accL ++ [x]) (accR ++ [y])
doSplit (x:[]) accL accR = (accL ++ [x], accR)
doSplit _ accL accR = (accL, accR)


split list = doSplit list [] []

l1 = [10, 20, 30, 40, 50, 60, 70, 80, 90]
(l2, l3) = split(l1)

main = do
    print l2
    print l3